lnwire/accept_channel.go

```diff
@@ -145,18 +144,28 @@ func (a *AcceptChannel) Decode(r io.Reader, pver uint32) error {
 		&a.DelayedPaymentPoint,
 		&a.HtlcPoint,
 		&a.FirstCommitmentPoint,
+		&a.ExtraData,
 	)
+}
+
+// UpfrontShutdownScript is the script to which the channel funds should be
+// paid when mutually closing the channel. This field is optional, and and has
+// a length prefix, so a zero will be written if it is not set and its length
+// followed by the script will be written if it is set.
+func (a *AcceptChannel) UpfrontShutdownScript() (DeliveryAddress, error) {
+	var addr DeliveryAddress
+
+	tlvs, err := a.ExtraData.ExtractRecords(addr.NewRecord())
```

Comment:
Related to the comment above, we could have the callers use methods to "attach" these fields, which exposes less details to them, and is an alternative to setting `nil` values. I still loosely favor letting the callers set the fields themselves, and hiding everything behind `Encode/Decode` re if it's actually a TLV or not. As we evolve the messages, I imagine we'll have more and more "TLV native" fields, which can start to clutter the call sites if they need to always manually pack/encode them.